pal/samples/esp8266/iothub_ll_telemetry_sample/iothub_ll_telemetry_sample.ino

```diff
@@ -0,0 +1,256 @@
+#ifdef SAMPLE_MQTT
+    IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = MQTT_Protocol;
+#endif // SAMPLE_MQTT
+#ifdef SAMPLE_HTTP
+   IOTHUB_CLIENT_TRANSPORT_PROVIDER protocol = HTTP_Protocol;
+#endif // SAMPLE_HTTP
+
+IOTHUB_DEVICE_CLIENT_LL_HANDLE device_ll_handle;
+
+static int callbackCounter;
+int receiveContext = 0;
+
+/* -- receive_message_callback --
+ * Callback method which executes upon receipt of a message originating from the IoT Hub in the cloud. 
+ * Note: Modifying the contents of this method allows one to command the device from the cloud. 
+ */
+static IOTHUBMESSAGE_DISPOSITION_RESULT receive_message_callback(IOTHUB_MESSAGE_HANDLE message, void* userContextCallback)
+{
+    int* counter = (int*)userContextCallback;
+    const unsigned char* buffer;
+    size_t size;
+    const char* messageId;
+
+    // Message properties
+    if ((messageId = IoTHubMessage_GetMessageId(message)) == NULL)
+    {
+        messageId = "<null>";
+    }
+
+    // Message content
+    if (IoTHubMessage_GetByteArray(message, (const unsigned char**)&buffer, &size) != IOTHUB_MESSAGE_OK)
+    {
+        LogInfo("unable to retrieve the message data\r\n");
+    }
+    else
+    {
+        LogInfo("Received Message [%d]\r\n Message ID: %s\r\n Data: <<<%.*s>>> & Size=%d\r\n", *counter, messageId, (int)size, buffer, (int)size);
+        // If we receive the work 'quit' then we stop running
+        if (size == (strlen("quit") * sizeof(char)) && memcmp(buffer, "quit", size) == 0)
+        {
+            g_continueRunning = false;
+        }
+    }
+
+    /* Some device specific action code goes here... */
+    (*counter)++;
+    return IOTHUBMESSAGE_ACCEPTED;
+}
+
+
+/* -- send_confirm_callback --
+ * Callback method which executes upon confirmation that a message originating from this device has been received by the IoT Hub in the cloud.
+ */
+static void send_confirm_callback(IOTHUB_CLIENT_CONFIRMATION_RESULT result, void* userContextCallback)
+{
+    (void)userContextCallback;
+    // When a message is sent this callback will get envoked
+    g_message_count_send_confirmations++;
+    LogInfo("Confirmation callback received for message %lu with result %s\r\n", (unsigned long)g_message_count_send_confirmations, MU_ENUM_TO_STRING(IOTHUB_CLIENT_CONFIRMATION_RESULT, result));
+}
+
+/* -- connection_status_callback --
+ * Callback method which executes on receipt of a connection status message from the IoT Hub in the cloud.
+ */
+static void connection_status_callback(IOTHUB_CLIENT_CONNECTION_STATUS result, IOTHUB_CLIENT_CONNECTION_STATUS_REASON reason, void* user_context)
+{
+    (void)reason;
+    (void)user_context;
+    // This sample DOES NOT take into consideration network outages.
+    if (result == IOTHUB_CLIENT_CONNECTION_AUTHENTICATED)
+    {
+        LogInfo("The device client is connected to iothub\r\n");
+    }
+    else
+    {
+        LogInfo("The device client has been disconnected\r\n");
+    }
+}
+
+void setup() {
```

Comment:
>setup [](start = 5, length = 5)

I see this function is doing more then setup. Can we make it more apt name? 
For me it is doing setup and starting our demo.

Reply:
It has to be named setup. Arduino requires it. I agree that it is awkward that it both sets up and runs. I think it's possible to use a Boolean to have the demo run once in the loop function. This could make for a better design imo.